### PR TITLE
Drop support of key types other than string and Buffer

### PR DESCRIPTION
**Pull request by vweevers**
Background: #186.

Todo:

- [x] Test against https://github.com/Level/subleveldown/issues/64 (cc @MeirionHughes)
- [x] Test against https://github.com/Level/level-ttl/issues/68#issuecomment-480505112
- [x] Test against https://github.com/Level/abstract-leveldown/pull/310

Should also remove the need for https://github.com/Level/codec/pull/51.

### CODE REVIEW

**Comment by vweevers** [comment]
Does this warrant a prerelease? That hasn't proven itself useful in the past (on levelup, leveldown) mainly because there wasn't any feedback on prereleases.

**Comment by ralphtheninja** [comment]
> Does this warrant a prerelease? That hasn't proven itself useful in the past (on levelup, leveldown) mainly because there wasn't any feedback on prereleases.

I think we can skip that, because no feedback and also yolo.

**Comment by MeirionHughes** [comment]
with ` "memdown": "git://github.com/Level/memdown.git#drop-custom-key-types",`  the failing test in https://github.com/Level/subleveldown/tree/tests/buffer-race-issue passes. I can try hooking this up with my store db and switch it to bytewise just to be sure.

**Comment by vweevers** [comment]
The race issues in `level-ttl` make it hard to tell if it is now compatible with `memdown` (less tests are failing, basically) so I also added a small test here, of mixed use of string and Buffer keys.

TLDR; everything looks good.

**Comment by vweevers** [comment]
One open question: this PR fixes keys, but similar issues exist on values. For example, assume we have a `level-mem` db with `utf8` value encoding, then `await db.put('key', Buffer.from('foo')); await db.get('key')` yields a Buffer, but with `leveldown` we'd get a string.

Should we go all the way, and also drop custom value types?

**Comment by vweevers** [comment]
> Should we go all the way, and also drop custom value types?

I say yes, because it's another incompatibility that isn't easy to solve otherwise. And it's common to swap keys and values (e.g. for indexing) so they should behave largely the same (I wish I realized this sooner).

Will open a second PR.
